aws/resource_aws_servicecatalog_provisioning_artifact.go

```diff
@@ -0,0 +1,297 @@
+		},
+	}
+}
+
+func resourceAwsServiceCatalogProvisioningArtifactCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).scconn
+
+	parameters := make(map[string]interface{})
+	parameters["description"] = d.Get("description")
+	parameters["disable_template_validation"] = d.Get("disable_template_validation")
+	parameters["name"] = d.Get("name")
+	parameters["template_physical_id"] = d.Get("template_physical_id")
+	parameters["template_url"] = d.Get("template_url")
+	parameters["type"] = d.Get("type")
+
+	input := &servicecatalog.CreateProvisioningArtifactInput{
+		IdempotencyToken: aws.String(resource.UniqueId()),
+		Parameters:       expandProvisioningArtifactParameters(parameters),
+		ProductId:        aws.String(d.Get("product_id").(string)),
+	}
+
+	if v, ok := d.GetOk("accept_language"); ok {
+		input.AcceptLanguage = aws.String(v.(string))
+	}
+
+	var output *servicecatalog.CreateProvisioningArtifactOutput
+	err := resource.Retry(iamwaiter.PropagationTimeout, func() *resource.RetryError {
+		var err error
+
+		output, err = conn.CreateProvisioningArtifact(input)
+
+		if tfawserr.ErrMessageContains(err, servicecatalog.ErrCodeInvalidParametersException, "profile does not exist") {
+			return resource.RetryableError(err)
+		}
+
+		if err != nil {
+			return resource.NonRetryableError(err)
+		}
+
+		return nil
+	})
+
+	if tfresource.TimedOut(err) {
+		output, err = conn.CreateProvisioningArtifact(input)
+	}
+
+	if err != nil {
+		return fmt.Errorf("error creating Service Catalog Provisioning Artifact: %w", err)
+	}
+
+	if output == nil || output.ProvisioningArtifactDetail == nil || output.ProvisioningArtifactDetail.Id == nil {
+		return fmt.Errorf("error creating Service Catalog Provisioning Artifact: empty response")
+	}
+
+	d.SetId(tfservicecatalog.ProvisioningArtifactID(aws.StringValue(output.ProvisioningArtifactDetail.Id), d.Get("product_id").(string)))
+
+	// Active and Guidance are not fields of CreateProvisioningArtifact but are fields of UpdateProvisioningArtifact.
+	// In order to set these to non-default values, you must create and then update.
+
+	return resourceAwsServiceCatalogProvisioningArtifactUpdate(d, meta)
+}
+
+func resourceAwsServiceCatalogProvisioningArtifactRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).scconn
+
+	artifactID, productID, err := tfservicecatalog.ProvisioningArtifactParseID(d.Id())
+
+	if err != nil {
+		return fmt.Errorf("error parsing Service Catalog Provisioning Artifact ID (%s): %w", d.Id(), err)
+	}
+
+	output, err := waiter.ProvisioningArtifactReady(conn, artifactID, productID)
+
+	if !d.IsNewResource() && tfawserr.ErrCodeEquals(err, servicecatalog.ErrCodeResourceNotFoundException) {
+		log.Printf("[WARN] Service Catalog Provisioning Artifact (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	if err != nil {
+		return fmt.Errorf("error describing Service Catalog Provisioning Artifact (%s): %w", d.Id(), err)
```

Comment:
since `ProvisioningArtifactStatus` returns a similar detailed message, which then gets returned from `waiter.ProvisioningArtifactReady`,  this log might be a bit redundant . personally i like returning the whole error in the status functions and then add context within the resource CRUD ops but either way works and no official convention exists i think, so as long just one copy of this sentence gets returned, sounds good to me 👍

Reply:
sounds good. we should come up with a convention.